### PR TITLE
Add regex matcher

### PR DESCRIPTION
**Pull request by makoto**
Added regex matcher at `with`.

Please sign off @joshuafleck @PeterWuMC

### CODE REVIEW

**Comment by joshuafleck** [comment]
Hey @makoto - what is the problem you are trying to solve by introducing this functionality to the matcher? Could you please provide an example of how you want to use this?

**Comment by makoto** [comment]
@joshuafleck I wanted to use it at https://github.com/simplybusiness/close_encounter/blob/alt-number/spec/features/step_definitions/close_encounter_steps.rb#L96 .

**Comment by joshuafleck** [comment]
Hey @makoto - thank you for the example. Now that I understand the problem you are wanting to solve, I'd like to suggest an alternate approach. I think that allowing regexp matches might provide a false sense of safety as it does not require the value you are matching to be tied to any specific key. What would you think of adding a new chainable method (e.g. `including`) that allows for a partial payload match? Something like:

```ruby
  expect { blah }
    .to publish_event('task')
    .including(key => value)
  end
```

**Comment by makoto** [comment]
Ideally it would be nicer for `publish_event` to take block passing actual argument like this 

```
expect { blah }
  .to publish_event('task') do | expected|
    expect(expected['key']).to eq value
  end
end
```
This way you can do anything you want inside the block and eliminates the need to introduce new method for every need. I think rspec's `to_receive` works in the same way. PeterWu and I tried to do that last friday and gave up :-(

**Comment by joshuafleck** [comment]
> PeterWu and I tried to do that last friday and gave up :-(

Let me know if I can help, @makoto !

File: spec/lib/rabbit_feed/testing_support/rspec_matchers/publish_event_spec.rb [hunk 1]
```diff
@@ -74,6 +74,43 @@ module RSpecMatchers
...
+            }.to publish_event(event_name, nil) do |actual_payload|
+              expect(actual_payload['field']).to eq 'value'
+            end
+          end
+
+          it 'validates block with `with` will be ignored' do
+            eval_block  = Proc.new {|actual_payload|
+              expect(actual_payload['field']).to eq 'value'
+            }
+            matcher = described_class.new(event_name, nil).with({filed: 'different name'})
```
Comment (joshuafleck) [inline spec/lib/rabbit_feed/testing_support/rspec_matchers/publish_event_spec.rb]: `filed` -> `field`?

File: spec/lib/rabbit_feed/testing_support/rspec_matchers/publish_event_spec.rb [hunk 2]
```diff
@@ -74,6 +74,43 @@ module RSpecMatchers
...
+            block   = Proc.new { RabbitFeed::Producer.publish_event event_name, {'field' => 'value'} }
+
+            (matcher.matches? block, &eval_block).should be true
+          end
+
+          it 'does not evaluates block with {}' do
+            # This is because block will dropped as part of transpforming `publish_event` to rspec custom helper
```
Comment (joshuafleck) [inline spec/lib/rabbit_feed/testing_support/rspec_matchers/publish_event_spec.rb]: `transpforming` -> `transforming`

**Comment by joshuafleck** [comment]
Looking good 👍 

signed-off-by: @joshuafleck

**Pull request merged**